packages/ui/src/box/box.tsx

```diff
@@ -31,44 +31,72 @@ const capitalize = ( str: string ): string =>
  * Converts a size value to a CSS design token property reference (with
  * fallback) or a calculated value based on the base unit.
  *
- * @param property The CSS property name.
- * @param target   The design system token target.
- * @param value    The size value, either a number (multiplier of base unit) or a string (token name).
- * @return A CSS value string with variable references.
+ * @param property    The CSS property name.
+ * @param target      The design system token target.
+ * @param value       The size value, either a number (multiplier of base unit) or a string (token name).
+ * @param offsetValue Optional CSS value to subtract (e.g., border width).
+ * @return            A CSS value string with variable references.
  */
 const getSpacingValue = (
 	property: string,
 	target: string,
-	value: number | string
+	value: number | string,
+	offsetValue?: string
+): string => {
+	const baseExpression =
+		typeof value === 'number'
+			? `var(--wpds-dimension-base) * ${ value }`
+			: `var(--wpds-dimension-${ property }-${ target }-${ value }, var(--wpds-dimension-${ property }-surface-${ value }))`;
+
+	if ( offsetValue ) {
+		return `calc(${ baseExpression } - ${ offsetValue })`;
+	}
+
+	return typeof value === 'number'
+		? `calc(${ baseExpression })`
+		: baseExpression;
+};
+
+const getBorderWidthValue = (
+	target: string,
+	borderWidth: Exclude< NonNullable< BoxProps[ 'borderWidth' ] >, number >
 ): string =>
-	typeof value === 'number'
```

Comment:
This is a good reminder we need to remove this `number` support exhaustively from paddings, similar to what we did with Stack gaps in #73852.